### ISSUE TITLE
Breaking API change between 4.0.1 and 4.0.2 in #findConfiguration()

### ISSUE THREAD

**Issue by egamma** [palantir/tslint#1789]
The interface `IConfigurationLoadResult` has changed between 4.0.1 and 4.0.2 in a breaking way.
The optional attribute `error` has been removed (see below). This changes the semantics of the function `findConfiguration`:
- In 4.0.2 `findConfiguration` throws an error when the configuration is invalid 
- in 4.0.1 `findConfiguration` doesn't throw an error when the configuration is invalid but returns the exception in the attribute `IConfigurationResult.error`.

`tslint\lib\configuration.d.ts` in version 4.0.1
```typescript
export interface IConfigurationLoadResult {
    error?: Error;
    path: string;
    results?: IConfigurationFile;
}
```

`tslint\lib\configuration.d.ts` in version 4.0.2
```typescript
export interface IConfigurationLoadResult {
    path: string;
    results?: IConfigurationFile;
}
```

**Comment by nchen63** [comment]
Sorry, my fault. Ironically, I made this change to revert an API breaking change from 3.x -> 4.0, which formerly threw an exception. Since the first API change was not intentional, I didn't think it warranted a major version change. 

At this point, it's not clear to me that the right thing to do would be to release a major version or to revert to previous behavior since 4.0.2 is out in the wild.

**Comment by egamma** [comment]
>Ironically, I made this change to revert an API breaking change from 3.x -> 4.0, which formerly threw an exception. 

Actually the this change does not fully revert the breaking change related to `findConfiguration`. In 4.0 the configuration is returned in the 'results` property of the returned object.

FYI, this is the related code that can handle 3.x, 4.0.1, and 4.0.2. It works and has been tested. So doing nothing would be preferred over creating another API swirl.

```typescript
function getConfiguration(filePath: string, configFileName: string): any {
     // ....
	let isDefaultConfig = false;
	let configuration;

	if (isTsLint4) {
		if (linterConfiguration.findConfigurationPath) {
			isDefaultConfig = linterConfiguration.findConfigurationPath(configFileName, filePath) === undefined;
		}
		let configurationResult = linterConfiguration.findConfiguration(configFileName, filePath);

		// between tslint 4.0.1 and tslint 4.0.2 the attribute 'error' has been removed from IConfigurationLoadResult
		// in 4.0.2 findConfiguration throws an exception as in version ^3.0.0
		if ((<any>configurationResult).error) {
			throw (<any>configurationResult).error;
		}
		configuration= configurationResult.results;
	} else {
		// prior to tslint 4.0 the findconfiguration functions where attached to the linter function
		if (linter.findConfigurationPath) {
			isDefaultConfig = linter.findConfigurationPath(configFileName, filePath) === undefined;
		}
		configuration = linter.findConfiguration(configFileName, filePath);
	}
        //...
```

**Comment by nchen63** [comment]
Thanks for the code excerpt. Closing the ticket since there is nothing to do at this point.
